common/test/acceptance/tests/lms/test_library.py

```diff
@@ -160,9 +182,122 @@ def test_shows_all_if_max_set_to_greater_value(self):
+            ),
+        )
+
+    @property
+    def _problem_headers(self):
+        """ Expected XBLock headers according to populate_library_fixture """
+        return frozenset(child.display_name.upper() for child in self.library_fixture.children)
+
+    def _set_library_content_settings(self, count=1, capa_type="Any Type"):
+        """
+        Sets library content XBlock parameters, saves, publishes unit, goes to LMS unit page and
+        gets children XBlock headers to assert against them
+        """
+        self._change_library_content_settings(count=count, capa_type=capa_type)
+        self._auto_auth(self.USERNAME, self.EMAIL, False)
+        self._goto_library_block_page()
+        return self.library_content_page.children_headers
+
+    def test_problem_type_selector(self):
+        """
+        Scenario: Ensure setting "Any Type" for Problem Type does not filter out Problems
+        Given I have a library with two "Select Option" and two "Choice Group" problems, and a course containing
+               LibraryContent XBlock configured to draw XBlocks from that library
+        When I set library content xblock Problem Type to "Any Type" and Count to 3 and publish unit
+        When I go to LMS courseware page for library content xblock as student
+        Then I can see 3 xblocks from the library of any type
+        When I set library content xblock Problem Type to "Choice Group" and Count to 1 and publish unit
+        When I go to LMS courseware page for library content xblock as student
+        Then I can see 1 xblock from the library of "Choice Group" type
+        When I set library content xblock Problem Type to "Select Option" and Count to 2 and publish unit
+        When I go to LMS courseware page for library content xblock as student
+        Then I can see 2 xblock from the library of "Select Option" type
+        When I set library content xblock Problem Type to "Matlab" and Count to 2 and publish unit
+        When I go to LMS courseware page for library content xblock as student
+        Then I can see 0 xblocks from the library
+        """
+        children_headers = self._set_library_content_settings(count=3, capa_type="Any Type")
+        self.assertEqual(len(children_headers), 3)
+        self.assertLessEqual(children_headers, self._problem_headers)
+
+        # Choice group test
+        children_headers = self._set_library_content_settings(count=1, capa_type="Multiple Choice")
+        self.assertEqual(len(children_headers), 1)
+        self.assertLessEqual(
+            children_headers,
+            set([header.upper() for header in ["Problem Choice Group 1", "Problem Choice Group 2"]])
+        )
+
+        # Choice group test
+        children_headers = self._set_library_content_settings(count=2, capa_type="Dropdown")
+        self.assertEqual(len(children_headers), 2)
+        self.assertLessEqual(
```

Comment:
Should be exactly equal in this case, no? Or is it possible that you could get the same problem twice?